### PR TITLE
Document new options to control resolution of native stack symbols

### PR DESCRIPTION
**Pull request by Sreekala-Gopakumar**
https://github.com/eclipse-openj9/openj9-docs/issues/977

-XX:[+|-]ShowNativeStackSymbols option added
Made changes to related topics

Signed-off-by: Sreekala Gopakumar <sreekala.gopakumar@ibm.com>

### CODE REVIEW

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
@keithc-ca - I have documented the new options introduced to control resolution of native stack symbols. Please check and confirm. Thanks!

File: docs/version0.35.md [hunk 1]
```diff
@@ -45,6 +46,12 @@ For threads that are waiting for a class initialization lock, the Java dump outp
 
 For more information, see [Threads](dump_javadump.md#threads).
 
+### New `-XX:[+|-]ShowNativeStackSymbols` option added
+
+This option controls whether Java dumps show the function names of the native stack symbols in native call stacks.
```
Comment (keithc-ca) [inline docs/version0.35.md:51]: This seems awkward to me; I suggest:

> This option controls whether Java dumps show the names of functions in native call stacks.

File: docs/xxshownativestacksymbols.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] http://openjdk.java.net/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# -XX:\[+|-\]ShowNativeStackSymbols
+
+This option controls whether Java&reg; dumps show the function names of the native stack symbols in native call stacks.
```
Comment (keithc-ca) [inline docs/xxshownativestacksymbols.md:27]: Again, suggest:

> This option controls whether Java&reg; dumps show the names of functions in native call stacks.

File: docs/xxshownativestacksymbols.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+## Syntax
+
+        -XX:-ShowNativeStackSymbols
+        -XX:+ShowNativeStackSymbols=<value>
+
+| Setting                    | Value  |Effect|Default|
+|----------------------------|--------|------|:-------:|
+|`-XX:-ShowNativeStackSymbols`|       |Don't show native stack symbols in response to a signal, such as a fault signal      |   |
+|`-XX:+ShowNativeStackSymbols`| basic| Show easily acquired native stack symbols only     | :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span>|
+|`-XX:+ShowNativeStackSymbols`| all| Show all the available native stack symbols     |   |
```
Comment (keithc-ca) [inline docs/xxshownativestacksymbols.md:38]: Other tables like this have spaces around the content in each cell, for example:
```
| `-XX:+ShowHiddenFrames`  | Enable  |                                                                                      |
```

File: docs/xxshownativestacksymbols.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+This option controls whether Java&reg; dumps show the function names of the native stack symbols in native call stacks.
+
+## Syntax
+
+        -XX:-ShowNativeStackSymbols
+        -XX:+ShowNativeStackSymbols=<value>
+
+| Setting                    | Value  |Effect|Default|
+|----------------------------|--------|------|:-------:|
+|`-XX:-ShowNativeStackSymbols`|       |Don't show native stack symbols in response to a signal, such as a fault signal      |   |
```
Comment (keithc-ca) [inline docs/xxshownativestacksymbols.md:36]: This isn't quite right. In `basic` mode, all symbols are shown for a faulting or signaled thread. With this option no native symbols are displayed, not even for a faulting or signaled thread; suggest:

> Don't show native stack symbols, not even in response to a signal, such as a fault signal

File: docs/xxshownativestacksymbols.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+|`-XX:+ShowNativeStackSymbols`| basic| Show easily acquired native stack symbols only     | :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span>|
```
Comment (keithc-ca) [inline docs/xxshownativestacksymbols.md:37]: Again, faulting threads are treated differently; suggest:
> Show all the available native stack symbols for a signaled or faulting thread, but only easily acquired native stack symbols for other threads

File: docs/xxshownativestacksymbols.md (hunk 3 above)
Comment (keithc-ca) [inline docs/xxshownativestacksymbols.md:38]: Suggest:
> Show all the available native stack symbols for all threads

File: docs/xxshownativestacksymbols.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+|`-XX:+ShowNativeStackSymbols`| all| Show all the available native stack symbols     |   |
+
+## Explanation
+
+Java dumps take longer to produce when the native stack symbols are included. If your Java application has a lot of threads, you might get an incomplete Java dump or the Java dump process might hang. You can avoid this situation by reducing the number of native stack symbols in the Java dump or by omitting them altogether.
```
Comment (keithc-ca) [inline docs/xxshownativestacksymbols.md:42]: I don't think it's appropriate to talk about the process hanging here; that's just a bug that should be fixed by eclipse/omr#6685.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
@keithc-ca - Incorporated the suggested changes. Please check. Thanks!

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
@keithc-ca - Can we close this issue, Keith? Any more changes? Thanks!

File: docs/xxshownativestacksymbols.md (hunk 3 above)
Comment (keithc-ca) [reply docs/xxshownativestacksymbols.md]: This suggestion has not been adopted. Please do, unless you have a good reason it should not be done.
Then, please rebase and squash.

Comment (Sreekala-Gopakumar) [reply docs/xxshownativestacksymbols.md]: Hi Keith, I checked the other topic tables and saw that the spaces in the cells are not consistent across. Also, I could see that in the generated content, the spaces in this table are same as the spaces in the other topic tables. Consistency in the spacing is being taken care of automatically at the time of generation irrespective of the number of spaces we put in the topic? It is with that belief that I didn't change the number of spaces in all the cells.

Comment (keithc-ca) [reply docs/xxshownativestacksymbols.md]: I looked at more than twenty files and the vast majority have spaces separating cell contents from the column boundaries (`|`); please do add those spaces.

Comment (Sreekala-Gopakumar) [reply docs/xxshownativestacksymbols.md]: As advised, I have added the spaces. Hope it is fine now. I have also squashed the commits.
Thanks!

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

File: docs/xxshownativestacksymbols.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+## Syntax
+
+        -XX:-ShowNativeStackSymbols
+        -XX:+ShowNativeStackSymbols=<value>
+
+| Setting                    | Value  |Effect|Default|
+|----------------------------|--------|------|:-------:|
+| `-XX:-ShowNativeStackSymbols` |       | Don't show native stack symbols, not even in response to a signal, such as a fault signal      |   |
+| `-XX:+ShowNativeStackSymbols` | basic | Show all the available native stack symbols for a signaled or faulting thread, but only easily acquired native stack symbols for other threads     | :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span>|
+| `-XX:+ShowNativeStackSymbols` | all | Show all the available native stack symbols for all threads     |   |
```
Comment (keithc-ca) [inline docs/xxshownativestacksymbols.md:38]: Please insert the spaces requested, for example
```
| Setting                    | Value  |Effect|Default|
```
should be
```
| Setting                    | Value  | Effect | Default |
```
and on the "basic" line, there should be a space before the final `|`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: docs/xxshownativestacksymbols.md (hunk 7 above)
Comment (Sreekala-Gopakumar) [reply docs/xxshownativestacksymbols.md]: Done. Please check. Thanks!

**Comment by keithc-ca** [comment]
jenkins doc stage

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**